### PR TITLE
owner and foreign api support

### PR DESCRIPTION
**Pull request by ravidio**
The PR adds foreign and owner API support to wallet713. Both API conform to default grin's wallet ones, and add additional functionality:
1. new route for `receive_invoice_tx` for foreign API
2. new method for `issue_send_tx` for owner API: `grinbox`

### CODE REVIEW

File: src/common/config.rs [hunk 1]
```diff
@@ -160,6 +167,22 @@ impl Wallet713Config {
             }
         }
     }
+
+    pub fn owner_api_address(&self) -> String {
+        self.owner_api_address.as_ref().map(|a| a.clone()).unwrap_or_else(|| String::from("127.0.0.1:12233"))
```
Comment (jaspervdm) [inline src/common/config.rs]: Maybe we should use the Grin port numbers as a default, and have a `1` in front for floonet

File: src/contacts/types.rs [hunk 2]
```diff
@@ -180,10 +180,10 @@ impl Display for KeybaseAddress {
 }
 
 pub fn version_bytes() -> Vec<u8> {
-    if is_mainnet() {
-        GRINBOX_ADDRESS_VERSION_MAINNET.to_vec()
-    } else {
+    if is_floonet() {
```
[the remark this answers is not in the document]
Comment (ravidio) [reply src/contacts/types.rs]: I did a `cargo update` and looks like the old `is_mainnet` is not there anymore?

File: docs/usage.md [hunk 3]
```diff
@@ -318,6 +319,40 @@ wallet713> $ invoice 10 --to @faucet -o 2
 
 Set corresponding `grin_node_uri` and `grin_node_secret` in your `~/.wallet713/XXX/wallet713.toml` where `XXX` is `floo` or `main` depending on which network you run the wallet for.
 
+## Configuring Foreign and Owner APIs
+
+Wallet713 provides a *variant* of grin's default wallet foregin and owner APIs.
```
[the remark this answers is not in the document]
Comment (ravidio) [reply docs/usage.md]: thx!

File: src/common/config.rs (hunk 1 above)
Comment (ravidio) [reply src/common/config.rs]: thx!

File: src/contacts/types.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ravidio) [reply src/contacts/types.rs]: Did another one now and it's back! Maybe I was imagining this :) in any case I reverted it to use is_mainnet now and updated dependencies.

**Pull request merged**